src/index.ts

```diff
@@ -39,36 +39,26 @@ export type Options = {
  * // When using 1, this will reject if the server isn't already running.
  * untilConnected({ target: 3000, maxRetries: 1 })
  */
-export function untilConnected(options: Options): Promise<void> {
+export async function untilConnected(options: Options): Promise<void> {
   const { port, host } = resolveConnectionOptions(options)
-  let maxRetries = options.maxRetries || 5
-  let retries = 0
+  const maxRetries = options.maxRetries || 5
 
-  function connectOrRetry(): Promise<void> {
-    retries++
-
-    return connect(port, host).catch(async (connectionError) => {
-      if (retries === maxRetries) {
-        const error = new Error(
-          `Failed to await connection at ${
-            host || ''
-          }:${port}. Connection never established (retries: ${retries}).`,
-          { cause: connectionError }
-        )
-        throw error
-      }
-
-      if (options.connectionInterval) {
-        await new Promise<void>((resolve) => {
-          setTimeout(resolve, options.connectionInterval)
-        })
-      }
-
-      return connectOrRetry()
-    })
+  let connectionError = null
```

Comment:
Let's annotate this as `Error | undefined`, then we can also drop the `null` default value.

Reply:
This comes at a cost of type asserting `connectionError = error as Error`